firecrest/AsyncExternalStorage.py

```diff
@@ -249,5 +249,5 @@ async def finish_download(
             if isinstance(target_path, str) or isinstance(target_path, pathlib.Path)
             else nullcontext(target_path)
         )
-        with urllib.request.urlopen(url) as response, context as out_file:
+        with urllib.request.urlopen(url['url']) as response, context as out_file:
```

Comment:
Now, I understand there has been a change in `url` type since Version 1.13.0.
Instead, one can do this:

```suggestion
_url = url["url"] if self._client._api_version > Version("1.13.0") else url
with urllib.request.urlopen(_url) as response, context as out_file:
            shutil.copyfileobj(response, out_file)
```
I hope this is appropriate.

Reply:
Hi @khsrali , this indeed a bug, thanks for reporting it! I meant to use this in line 243:
```python
url = await self.object_storage_link
```
instead of `object_storage_data`

Reply:
Hi @ekouts , yes that actually makes sense :)
Anyways, apparently parenthesis was needed here: 

`            return (await self.object_storage_data)["url"]`

Otherwise, asyncio returns: TypeError: 'coroutine' object is not subscriptable

I just checked `ExternalDownload` it was already correct, cheers!

Reply:
> I just checked ExternalDownload it was already correct, cheers!

Oh right... I found very strange that this bug was not reported already! 😅